### ISSUE TITLE
Warning: createRoot(): Creating roots directly with document.body is discouraged

### ISSUE THREAD

**Issue by neilfranci** [wxt-dev/wxt#456]
### Describe the bug

When creating a content script with React it shows a warning:
`
react-dom.development.js:86 Warning: createRoot(): Creating roots directly with document.body is discouraged, since its children are often manipulated by third-party scripts and browser extensions. This may lead to subtle reconciliation issues. Try using a container element created for your app.
`

### To Reproduce

Share minimal reproduction. Examples of a minimal reproduction:
[temp.zip](https://github.com/wxt-dev/wxt/files/14309253/temp.zip)

Steps to reproduce the bug using the reproduction:

1. Install dependencies: `pnpm i`
2. Start dev mode: `pnpm dev`
3. Wait for the browser open and try to go to any website or just google
4. F12 and look at the console tab

### Screenshots

![image](https://github.com/wxt-dev/wxt/assets/66520542/8ef865b9-68c0-426b-b6f2-e6d1149acadb)

### Environment

```
  System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i5-12500H
    Memory: 4.58 GB / 15.73 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.15.3 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (121.0.2277.112)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    wxt: ^0.16.8 => 0.16.8
```

### Additional context

Everything still works fine, I think. But the warning is just annoying.

**Comment by aklinker1** [comment]
Yes, `container` is a `body` element. For CSS frameworks like tailwind to properly work, I mimic a regular document inside the ShadowRoot. You can see this by inspecting the DOM.

For the react example, I made the react example as simple as possible, but I can update it since it's a little misleading.

```ts
      onMount: (container) => {
-        const root = ReactDOM.createRoot(container);
+        const app = document.createElement('div');
+        container.append(app);
+        const root = ReactDOM.createRoot(app);
        root.render(<Main />);
        return root;
      },
```

**Comment by aklinker1** [comment]
Docs updated 👍
